### ISSUE TITLE
Consider restricting `awskms://` options for admission control

### ISSUE THREAD

**Issue by mattmoor** [sigstore/cosign#1930]
I was looking at the AWS KMS options here: https://github.com/sigstore/cosign/blob/main/KMS.md#aws

There are soooo many ways of expressing the same key, but some of these forms only partially specify things and lean on the environment (e.g. `AWS_REGION`) and I don't think we should support these forms in ClusterImagePolicy.

I think we should require one of the form(s): `awskms://[host]/{arn}` where:
1. `host` is the optional endpoint, and
2. `arn` is either the key ARN or an alias ARN.

@hectorj2f @vaikas WDYT?

**Comment by hectorj2f** [comment]
Yes, i am happy with restricting it. I also find it complex and we already had several bugs about it.
